src/libraries/System.Text.Json/tests/Serialization/ExceptionTests.cs

```diff
@@ -527,4 +527,100 @@ public static void UnsupportedTypeFromRoot()
             Assert.DoesNotContain("Path: $", ex.Message);
         }
     }
+
+    public class OpenGenericTests_Span : OpenGenericTests
+    {
+        public OpenGenericTests_Span() : base(SerializationWrapper.SpanSerializer) { }
+    }
+
+    public class OpenGenericTests_String : OpenGenericTests
+    {
+        public OpenGenericTests_String() : base(SerializationWrapper.StringSerializer) { }
+    }
+
+    public class OpenGenericTests_Stream : OpenGenericTests
+    {
+        public OpenGenericTests_Stream() : base(SerializationWrapper.StreamSerializer) { }
+    }
+
+    public class OpenGenericTests_StreamWithSmallBuffer : OpenGenericTests
+    {
+        public OpenGenericTests_StreamWithSmallBuffer() : base(SerializationWrapper.StreamSerializerWithSmallBuffer) { }
+    }
+
+    public class OpenGenericTests_Writer : OpenGenericTests
+    {
+        public OpenGenericTests_Writer() : base(SerializationWrapper.WriterSerializer) { }
+    }
+
+    public abstract class OpenGenericTests
+    {
+        private SerializationWrapper Serializer { get; }
+
+        public OpenGenericTests(SerializationWrapper serializer)
+        {
+            Serializer = serializer;
+        }
+
+        [Fact]
+        public void DeserializeOpenGeneric()
+        {
+            foreach (object[] obj in TypesWithOpenGenerics_ToDeserialize())
+            {
+                Type type = (Type)obj[0];
+
+                InvalidOperationException ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize("", type));
+                Assert.Contains(type.ToString(), ex.ToString());
+            }
+        }
+
+        [Fact]
+        public void SerializeOpenGeneric()
+        {
+            object obj;
+
+            foreach (object[] types in TypesWithOpenGenerics_ToSerialize())
+            {
+                Type type = (Type)types[0];
+
+                if (type.GetGenericArguments().Length == 1)
+                {
+                    obj = Activator.CreateInstance(type.MakeGenericType(typeof(int)));
+                }
+                else
+                {
+                    obj = Activator.CreateInstance(type.MakeGenericType(typeof(string), typeof(int)));
+                }
+
+                ArgumentException ae = Assert.Throws<ArgumentException>(() => Serializer.Serialize(obj, type));
```

Comment:
Test also passing `null` instead of `obj`, that way you can actually trigger the new exception.